src/bls12_381/ec.rs

```diff
@@ -1420,6 +1510,39 @@ pub mod g2 {
             }
         }
 
+        fn get_swenc_const0() -> Fq2 {
+            Fq2 {
+                c0: super::super::fq::SWENC_CONST0,
+                c1: Fq::zero()
+            }
+        }
+
+        fn get_swenc_const1() -> Fq2 {
+            Fq2 {
+                c0: super::super::fq::SWENC_CONST1,
+                c1: Fq::zero()
+            }
+        }
+
+        /// Maps any elliptic curve point to the subgroup G2,
+        /// multiplying by the cofactor
+        fn scale_by_cofactor(&self) -> Self {
+            // Cofactor of G2 is 305502333931268344200999753193121504214466019254188142667664032982267604182971884026507427359259977847832272839041616661285803823378372096355777062779109.
+            // Calculated by: ((x**8) - (4 * (x**7)) + (5 * (x**6)) - (4 * (x**4)) + (6 * (x**3)) - (4 * (x**2)) - (4*x) + 13) // 9
+            // where x is the BLS parameter.
+            let bits = "101110101010100001110101001010101000001010011100111111100010000100100011101010100000111100100101000011101101010001000000010110011011001000111011110010001010100011100001000010110101011101010100110100010100010000001011011001011100101101001111101110111111010011000101000111100011100101101001101100111101000001011101111001000010101001101111110001010010011101001100110100100011010111000010110000101101110110001101110011110000110111100001100011100001100111100011100001110001110001100011100011100100011100011100101"
```

Comment:
Mention cheap hack to avoid inexpressible huge numeric literal and use binary multiplication.

Reply:
done!